### PR TITLE
Multiaccount SQS sensor capability

### LINKED ISSUE #83: Multiaccount SQS sensor support
Stackstorm can access only queues from the same region and account. With STS AssumeRole it is possible to fetch messages from queues in other account(s) and region(s).  It's not always possible for the producer system to push messages cross-account (one example is CloudWatch, which can push only to queues in the same region and account).

### PR DESCRIPTION
**Pull request by stefangusa**
Fixes #83

### CODE REVIEW

**Review by blag — CHANGES_REQUESTED** [review]
Looking mostly good, but needs some improvements.

File: sensors/sqs_sensor.py [hunk 1]
```diff
@@ -87,8 +95,16 @@ def _get_config_entry(self, key, prefix=None):
         return value
 
     def _may_setup_sqs(self):
-        queues = self._get_config_entry(key='input_queues', prefix='sqs_sensor')
+        self.access_key_id = self._get_config_entry('aws_access_key_id')
+        self.secret_access_key = self._get_config_entry('aws_secret_access_key')
+        self.aws_region = self._get_config_entry('region')
```
[the remark this answers is not in the document]
Comment (stefangusa) [reply sensors/sqs_sensor.py:126]: This is the same code as the one existent now, just moved upwards, so I did not name these variables.

File: sensors/sqs_sensor.py [hunk 2]
```diff
@@ -87,61 +96,156 @@ def _get_config_entry(self, key, prefix=None):
...
+        aws_region = self.aws_region
+
+        # Netloc will be empty if the queue is just a name
+        if queue.netloc:
+            try:
+                account_id = queue.path.split('/')[1]
+            except IndexError as e:
+                six.reraise(type(e), type(e)(
+                    "Queue URL must contain the account ID as the first part of the path, "
+                    "eg: https://.../<account_id>"),
```
Comment (blag) [inline sensors/sqs_sensor.py]: Now I'm thinking we should be even more explicit:
```suggestion
                    "eg: https://sqs.<aws_region>.amazonaws.com/<account_id>/<queue_name>"),
```

File: sensors/sqs_sensor.py [hunk 3]
```diff
@@ -98,50 +114,117 @@ def _may_setup_sqs(self):
...
+
+        sqs_res = self._setup_sqs(self.sessions[account_id], account_id, region)
+        if sqs_res is None:
+            return None
 
-    def _get_queue_by_name(self, queueName):
-        ''' Fetch QUEUE by it's name create new one if queue doesn't exist '''
         try:
-            return self.sqs_res.get_queue_by_name(QueueName=queueName)
+            if self._check_queue_if_url(queue):
```
[the remark this answers is not in the document]
Comment (blag) [reply sensors/sqs_sensor.py]:
```python
            self.input_queues = [six.moves.urllib.parse.urlparse(queue) for queue in queues]
```

Good compromise. :+1:

**Review by blag — APPROVED** [review]
One last tweak to exception message, but I think this is looking good.

File: sensors/sqs_sensor.py [hunk 4]
```diff
@@ -87,61 +96,156 @@ def _get_config_entry(self, key, prefix=None):
...
+
+        # Netloc will be empty if the queue is just a name
+        if queue.netloc:
+            try:
+                account_id = queue.path.split('/')[1]
+            except IndexError as e:
+                six.reraise(type(e), type(e)(
+                    "Queue URL must contain the account ID as the first part of the path, "
+                    "eg: https://.../<account_id>"),
+                    sys.exec_info()[2])
```
Comment (blag) [inline sensors/sqs_sensor.py]:
```suggestion
                    sys.exc_info()[2])
```

File: sensors/sqs_sensor.py [hunk 5]
```diff
@@ -87,61 +96,156 @@ def _get_config_entry(self, key, prefix=None):
...
+            else:
+                self._logger.debug("Using %s as account_id", account_id)
+
+            try:
+                aws_region = queue.netloc.split('.')[1]
+            except IndexError:
+                six.reraise(type(e), type(e)(
+                    "Queue URL must contain the AWS region, "
+                    "eg: https://sqs.<aws_region>.amazonaws.com/..."),
+                    sys.exec_info()[2])
```
Comment (blag) [inline sensors/sqs_sensor.py]:
```suggestion
                    sys.exc_info()[2])
```

**Review by blag — CHANGES_REQUESTED** [review]
Fixes for my own typos.

**Comment by stefangusa** [comment]
And renamed 'roles_arns' to 'roles'.

**Review by blag — APPROVED** [review]
Thanks for seeing this through!

**Review by nmaludy — APPROVED** [review]
LGTM!

**Review by nmaludy — CHANGES_REQUESTED** [review]
Sorry two small changes.

Can you please add a CHANGELOG entry in: https://github.com/StackStorm-Exchange/stackstorm-aws/blob/master/CHANGES.md

And also bump the version in `pack.yaml` (https://github.com/StackStorm-Exchange/stackstorm-aws/blob/master/pack.yaml#L22) by 0.1  to be `1.3.0`.

Please and thank you!

After these two things i'll gladly merge!

File: CHANGES.md [hunk 6]
```diff
@@ -1,5 +1,8 @@
 # Changelog
 
+## 1.2.4
```
Comment (nmaludy) [inline CHANGES.md]: FYI the version here doesn't match the version in pack.yaml :( sorry!!!

```suggestion
## 1.3.0
```
You should be able to accept this suggestion and have it auto-commit the change.

**Review by nmaludy — APPROVED** [review]
LGTM

**Pull request merged**